### PR TITLE
UI: Handle missing TrialTemplates

### PR DESCRIPTION
**Pull request by kimwnasptd**
closes https://github.com/kubeflow/katib/issues/1620

This is also part of the 0.12 release effort https://github.com/kubeflow/katib/issues/1597

The UI should gracefully handle the case of missing TrialParameters. 
I changed the UI to use `yaml` input by default and if no TrialTemplates were found then the `ConfigMap` option would be disabled. If the TrialTemplates were fetched correctly, and are not an empty list, then the UI will select the `ConfigMap` option.

Screenshot from 2021-09-01 19-30-43

**Checklist:**

- [ ] [Docs](https://www.kubeflow.org/docs/components/katib/) included if any changes are user facing

/cc @andreyvelich 
/assign @kimwnasptd

### CODE REVIEW

File: pkg/new-ui/v1beta1/frontend/src/app/pages/experiment-creation/trial-template/trial-template.component.html [hunk 1]
```diff
@@ -37,7 +37,13 @@
     <mat-form-field appearance="outline" class="fit-content">
       <mat-label>Source Type</mat-label>
       <mat-select formControlName="type">
-        <mat-option value="configmap">ConfigMap</mat-option>
+        <mat-option
+          [disabled]="!templates.length"
+          [matTooltip]="!templates.length ? 'No TrialTemplates found' : ''"
```
Comment (andreyvelich) [inline pkg/new-ui/v1beta1/frontend/src/app/pages/experiment-creation/trial-template/trial-template.component.html:42]: Let's keep it separate
```suggestion
          [matTooltip]="!templates.length ? 'No Trial templates found' : ''"
```

**Review by andreyvelich — COMMENTED** [review]
Thank you for this fix @kimwnasptd.
Just small comment from me.

File: pkg/new-ui/v1beta1/frontend/src/app/pages/experiment-creation/trial-template/trial-template.component.html (hunk 1 above)
Comment (kimwnasptd) [reply pkg/new-ui/v1beta1/frontend/src/app/pages/experiment-creation/trial-template/trial-template.component.html:42]: Makes sense!

**Review by andreyvelich — COMMENTED** [review]
/lgtm
/approve

@kimwnasptd Please can you cherry-pick this commit on release-0.12 branch ?

**Pull request merged**